onboarding/source/amb_light_service.c

```diff
@@ -35,19 +48,86 @@ obc_error_code_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here. Return error code if task/queue was not created successfully.
 
+    if (lightServiceTaskHandle == NULL) {
+        lightServiceTaskHandle = xTaskCreateStatic(
+                                                lightServiceTask,
+                                                LIGHT_SERVICE_NAME,
+                                                LIGHT_SERVICE_STACK_SIZE,
+                                                NULL,
+                                                LIGHT_SERVICE_PRIORITY,
+                                                lightServiceTaskStack,
+                                                &lightServiceTaskBuffer);
+    }
+
+    if (lightServiceTaskHandle == NULL) {
+        return OBC_ERR_CODE_TASK_CREATION_FAILED;
+    }
+
+    if (lightServiceQueueHandle == NULL) {
+        lightServiceQueueHandle = xQueueCreateStatic(
+                                                    LIGHT_SERVICE_QUEUE_LENGTH,
+                                                    LIGHT_SERVICE_ITEM_SIZE,
+                                                    lightServiceQueueStorage,
+                                                    &lightServiceQueue);
+    }
+
+    if (lightServiceQueueHandle == NULL)
+        return OBC_ERR_CODE_QUEUE_CREATION_FAILED;
+
+    return OBC_ERR_CODE_SUCCESS;
+
     /* USER CODE END */
 }
 
+uint32_t getLightSensorData(void) {
```

Comment:
Great job making a helper function for this!

It'd make sense to declare this a `static` function. That would make it accessible only to functions in the same compilation unit. A compilation unit is _basically_ just a source file with the content of its included header files copied into it. You can read more about them [here](https://www.cs.auckland.ac.nz/references/unix/digital/AQTLTBTE/DOCU_015.HTM)). For example, a static function in this `.c` source file would only be accessible by other functions in this file.

Functions, by default (i.e. without the `static` keyword), are accessible in other compilation units. However, if this is your intention, you'd typically put its function declaration in a header file (so `amb_light_service.h` in this case).